### PR TITLE
Fixing code to run with v1 grades agorithm

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?
N/A
#### What's this PR do?
Fixed the code to work with the `v1` grade algorithm

Run the tests with the environment variable:
`FEATURE_FINAL_GRADE_ALGORITHM=v0`
and
`FEATURE_FINAL_GRADE_ALGORITHM=v1`

The tests should pass in both cases.

### CODE REVIEW

File: dashboard/utils_test.py [hunk 1]
```diff
@@ -645,6 +653,43 @@ def test_count_courses_passed_fa(self):
         )
         assert mmtrack.count_courses_passed() == 1
 
+    @override_settings(FEATURES={"FINAL_GRADE_ALGORITHM": "v1"})
+    def test_count_courses_passed_normal(self):
```
Comment (rhysyngsun) [inline dashboard/utils_test.py]: Is it worth suffixing this test with `_v1` just like the others or is your thought that we'll eventually sunset the `_v0` code and you don't want to have to rename everything down the road?

File: grades/factories.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+FAKE = faker.Factory.create()
+
+
+class FinalGradeFactory(DjangoModelFactory):
+    """Factory for FinalGrade"""
+    user = SubFactory(UserFactory)
+    course_run = SubFactory(CourseRunFactory)
+    grade = randint(30, 100)/100.0
+    passed = FAKE.boolean()
```
Comment (rhysyngsun) [inline grades/factories.py]: For both `grade` and `passed` here, the values are random initialized staticly, is that the intent?

File: exams/utils.py [hunk 3]
```diff
@@ -41,8 +41,12 @@ def authorize_for_exam(mmtrack, course_run):
 
         # if user passed the course and currently not authorization for that run then give
         # her authorizations.
+        try:
+            passed = mmtrack.has_passed_course(course_run.edx_course_key)
+        except:  # pylint: disable=bare-except
+            return
```
Comment (rhysyngsun) [inline exams/utils.py]: Is there a reason we don't log the exception here? I worry about this bare exception hiding quiet failures which will make it difficult to debug.

**Comment by rhysyngsun** [comment]
Tests passed for me locally with the two `FEATURE_FINAL_GRADE_ALGORITHM` values. Code makes sense based on our discussion yesterday. A few questions above.

File: dashboard/utils_test.py (hunk 1 above)
Comment (giocalitri) [inline dashboard/utils_test.py]: I marked as `v0` the ones that should be removed later when we make `v1` default

File: grades/factories.py (hunk 2 above)
Comment (giocalitri) [inline grades/factories.py]: Not sure if I understood the question

File: exams/utils.py (hunk 3 above)
Comment (giocalitri) [inline exams/utils.py]: actually there is not: I will add it.

File: grades/factories.py (hunk 2 above)
Comment (rhysyngsun) [inline grades/factories.py]: The right operand is only evaluated once during the lifetime of the program, resulting in a fixed value across all calls to the factory. Example:

```python
In [8]: for u in User.objects.all()[:10]:
   ...:     g = FinalGradeFactory(user=u)
   ...:     print((g.passed, g.grade))
   ...: 
(True, 0.91)
(True, 0.91)
(True, 0.91)
(True, 0.91)
(True, 0.91)
(True, 0.91)
(True, 0.91)
(True, 0.91)
(True, 0.91)
(True, 0.91)

```

Is that intended?

Comment (giocalitri) [inline grades/factories.py]: no: this should be fixed

**Comment by giocalitri** [comment]
I think I addressed the comments

**Pull request merged**